From ff26d8d2af74ef07941dcfc9c016f17b7548fe05 Mon Sep 17 00:00:00 2001
From: Ryan Blue <blue@apache.org>
Date: Wed, 26 Aug 2015 12:51:49 -0700
Subject: [PATCH 098/140] KITE-1065: Add support for Oozie delegation tokens.

Oozie passes delegation tokens via HADOOP_TOKEN_FILE_LOCATION. This adds
support to pick up those tokens and add them to the DefaultConfiguration.
---
 .../org/kitesdk/data/spi/DefaultConfiguration.java |   14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/DefaultConfiguration.java b/kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/DefaultConfiguration.java
index e43a6b0..e3dd361 100644
--- a/kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/DefaultConfiguration.java
+++ b/kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/DefaultConfiguration.java
@@ -38,16 +38,22 @@ public class DefaultConfiguration {
   private static Configuration conf;
 
   static{
-    //read system property for Oozie Action Configuration XML
+    // read system property for Oozie Action Configuration XML
     String oozieActionXml = System.getProperty("oozie.action.conf.xml");
-    if(oozieActionXml != null && oozieActionXml.length() > 0) {
+    if (oozieActionXml != null && oozieActionXml.length() > 0) {
       Configuration.addDefaultResource(new Path("file:///", oozieActionXml).toString());
     }
 
     conf = new Configuration();
-  }
-
 
+    // check for Oozie delegation tokens
+    // based on Oozie's SqoopMain
+    String tokenFile = System.getenv("HADOOP_TOKEN_FILE_LOCATION");
+    if (tokenFile != null) {
+      conf.set("mapreduce.job.credentials.binary", tokenFile);
+      System.setProperty("mapreduce.job.credentials.binary", tokenFile);
+    }
+  }
 
   /**
    * Get a copy of the default Hadoop {@link Configuration}.
-- 
1.7.9.5

